### PR TITLE
vdaf: Merge `PrepareStep` functionality into `Aggregator`

### PR DESCRIPTION
**Pull request by cjpatton**
Merges the functionality provided by `PrepareStep` into `Aggregator`.
This allows long-lived VDAF context parameters to be used by the
prepare-state update algorithm (now `Aggregator::next()`). It also
simplifies the API slightly by removing a trait.

### CODE REVIEW

File: src/vdaf.rs [hunk 1]
```diff
@@ -115,8 +115,11 @@ pub trait Client: Vdaf {
 
 /// The Aggregator's role in the execution of a VDAF.
 pub trait Aggregator: Vdaf {
-    /// Initial state of the Aggregator during the Prepare process.
-    type PrepareInit: PrepareStep<Self::OutputShare>;
+    /// State of the Aggregator during the Prepare process.
+    type PrepStep;
+
+    /// The type of messages exchanged among the Aggregators during the Prepare process.
+    type PrepMessage;
```
Comment (tgeoghegan) [inline src/vdaf.rs:122]: nit:
```suggestion
    type PrepareStep;

    /// The type of messages exchanged among the Aggregators during the Prepare process.
    type PrepareMessage;
```

File: src/vdaf.rs [hunk 2]
```diff
@@ -126,76 +129,82 @@ pub trait Aggregator: Vdaf {
...
+    fn unshard<M: IntoIterator<Item = Self::AggregateShare>>(
+        &self,
+        agg_param: &Self::AggregationParam,
+        agg_shares: M,
+    ) -> Result<Self::AggregateResult, VdafError>;
 }
 
 /// A state transition of an Aggregator during the Prepare process.
-pub enum PrepareTransition<S: PrepareStep<O>, O> {
+pub enum PrepTransition<S, M, O> {
```
Comment (tgeoghegan) [inline src/vdaf.rs:205]:
```suggestion
pub enum PrepareTransition<S, M, O> {
```

**Review by tgeoghegan — APPROVED** [review]
I have a slight preference for type names like `PrepareStep` over `PrepStep` but it's not a dealbreaker and otherwise LGTM.

File: src/vdaf.rs (hunk 1 above)
Comment (cjpatton) [reply src/vdaf.rs]: I figured you might hate this :) Fixed, here and elsewhere.

**Comment by cjpatton** [comment]
Rebased and prefixed Prepare methods with "prepare_" (as discussed on Slack).

**Comment by cjpatton** [comment]
Silence clippy.

**Pull request merged**